framework/mongo/utils.py

```diff
@@ -63,10 +63,18 @@ def wrapper(cls):
 
 def get_or_http_error(Model, pk):
     instance = Model.load(pk)
+
     if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
+        if hasattr(instance, 'registrations'):
+            raise HTTPError(http.GONE, data=dict(
+                message_short="Project Deleted",
+                message_long="This Project has been deleted"
```

Comment:
Replied in https://github.com/CenterForOpenScience/osf.io/issues/3421#issuecomment-118998219